src/Estimators/src/BaseEstimatorFromFootIMU.cpp

```diff
@@ -170,100 +209,254 @@ bool BaseEstimatorFromFootIMU::advance()
+    double measuredYaw = m_measuredRPY(2);
 
-    // manif::SE3d pose matrix that employs: desired Position, measured Roll,
-    // measured Pitch, desired Yaw.
-    manif::SE3d T_foot_raw(m_desiredTranslation, m_measuredRotationCorrected);
+    // manif::SE3d pose matrix that employs: offset Position, measured Roll, measured Pitch, offset
+    // Yaw.
+    manif::SE3d T_foot_imu(m_noTras, m_measuredRotationCorrected);
+    manif::SE3d T_foot_tilt(m_noTras, m_measuredTilt);
+    manif::SE3d T_foot_offset(m_offsetTranslation, m_offsetRotationCasted);
 
-    // finding the positions of the foot corners in world frame given `T_foot_raw`
+    // finding the positions of the foot corners in world frame given `T_foot_imu`
     // pose matrix.
 
+    // resetting the vector of transformed foot corners from previous iteration.
+    m_tiltedFootCorners.clear();
+
     // for each corner we compute the position in the inertial frame
-    for (const auto& corner : m_cornersInLocalFrame)
+    for (const auto& corner : m_cornersInInertialFrame)
     {
-        m_transformedFootCorners.emplace_back(T_foot_raw.act(corner));
+        m_tiltedFootCorners.emplace_back(T_foot_tilt.act(corner));
     }
 
     // The center of the foot sole is at ground level --> some corners may be
-    // under ground level. The foot may need to be lifted by an offset value.
+    // under ground level. The foot may need to be translated by a 3D offset value.
 
     // extracting the vertical quotes of the foot corners and finding the lowest
     // corner.
-    auto lowestCorner = std::min_element(m_transformedFootCorners.begin(),
-                                         m_transformedFootCorners.end(),
-                                         [](const Eigen::Vector3d& a, const Eigen::Vector3d& b) {
-                                             return a[2] < b[2];
-                                         });
+    Eigen::VectorXd cornersZ;
+    cornersZ.resize(m_cornersInInertialFrame.size());
+    cornersZ.setZero();
+    int index = 0;
+    for (const auto& corner : m_tiltedFootCorners)
+    {
+        cornersZ[index] = corner[2];
+        index++;
+    }
 
     // finding the index of the lowest corner.
-    int supportCornerIndex = std::distance(m_transformedFootCorners.begin(), lowestCorner);
-
-    Eigen::Vector3d p_desired(0, 0, 0);
-    Eigen::Vector3d vertexOffset(0, 0, 0);
+    double minZ = cornersZ[0];
+    int indexMinZ = 0;
+    for (int i = 1; i < cornersZ.size(); i++)
+    {
+        if (cornersZ[i] < minZ)
+        {
+            minZ = cornersZ[i];
+            indexMinZ = i;
+        }
+    }
+    m_state.supportCornerIndex = indexMinZ;
 
-    // finding the offset vector needed to bring the lowest corner back to its
-    // desired position.
-    switch (supportCornerIndex)
+    // checking that the index of the lowest corner is within the range [0, 3].
+    if (!(0 <= m_state.supportCornerIndex <= 3))
```

Comment:
I would simplify the if state as 
```suggestion
    if (m_state.supportCornerIndex > 3 || m_state.supportCornerIndex < 0)
```